packages/patternfly-4/react-core/src/components/Form/FormGroup.js

```diff
@@ -0,0 +1,168 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from '@patternfly/patternfly-next/components/Form/form.css';
+import { ASTERISK } from '../../internal/htmlConstants';
+import { FormContext } from '../Form/FormContext';
+import { css, getModifier } from '@patternfly/react-styles';
+
+const childrenArrayContains = (childrenArray, attribute, errorMessage) => {
+  let throwError = true;
+  childrenArray.forEach(child => {
+    if (child.props[attribute]) {
+      throwError = false;
+    }
+  });
+  if (throwError) {
+    return new Error(errorMessage);
+  }
+  return null;
+};
+
+const propTypes = {
+  /** Anything that can be rendered as FormGroup content. */
+  children: props => {
+    if (typeof props.label !== 'string') {
+      if (Array.isArray(props.children)) {
+        return childrenArrayContains(
+          props.children,
+          'aria-label',
+          `When label is not a string, at least one of Form Group children should have an aria-label attribute.`
+        );
+      } else if (
+        typeof props.children === 'object' ||
+        (props.children.$$typeof && props.children.$$typeof === 'Symbol(react.element)')
+      ) {
+        if (!props.children.props['aria-label']) {
+          return new Error(`When label is not a string, a Form Group child should have an aria-label attribute.`);
+        }
+      }
+    } else if (Array.isArray(props.children)) {
+      return childrenArrayContains(props.children, 'id', `At least one of Form Group children should have an id.`);
+    } else if (!props.children.props.id) {
+      return new Error(`A child element of Form Group has to have an id.`);
+    }
+    return null;
+  },
+  /** Additional classes added to the FormGroup. */
+  className: PropTypes.string,
+  /** Label text before the field. */
+  label: PropTypes.oneOfType([PropTypes.string, PropTypes.node, PropTypes.func]),
+  /** Sets the FormGroup required. */
+  isRequired: PropTypes.bool,
+  /** Sets the FormGroup isValid. */
+  isValid: PropTypes.bool,
+  /** Sets the FormGroup isInline. */
+  inline: PropTypes.bool,
+  /** Helper text after the field. It can be a simple text, a function or an object. */
+  helperText: PropTypes.oneOfType([PropTypes.string, PropTypes.node, PropTypes.func]),
+  /** Helper text after the field when the field is isValid. It can be a simple text, a function or an object. */
+  helperTextInvalid: PropTypes.oneOfType([PropTypes.string, PropTypes.node, PropTypes.func]),
+  /** ID of the included field. It has to be the same for proper working. */
+  fieldId: props => {
```

Comment:
Initially I named this 'id' but it's not an id of the formGroup, but id of the some input field (used in 'for' in label). It would be a little bit confusing. However I am open to change 'field' to something else.